### DISCUSSION TITLE
Depending on where position in query is good idea?

### DISCUSSION THREAD

**Discussion by montyclt** [ravendb/ravendb#14028] [Q&A] [answered]
Hi!

I have this type of document:
```json
{
    "Name": "IVA",
    "Country": "ESP",
    "Region": null,
    "Percentage": 21
}
```
```json
{
    "Name": "IGIC",
    "Country": "ESP",
    "Region": "Canarias",
    "Percentage": 7
}
```

Then, I need a query that input country and region, get the Tax that applies to the country and the region, and if the region doesn't exists, get the tax that applies to the country and region is null.

I notice that the where position differs on the result, so I wrote this query:
```rql
from 'Core/Invoicing/TaxRule'
where Country == "ESP"
and Region == "Canarias"
or Region == null
limit 0, 1
```

The query (without limit) get **firsts** the tax that fit country and region, and then the tax that fit only the country. Limit to 1 returns me the result that I need.
![Screenshot 2022-04-07 at 14 44 03](https://user-images.githubusercontent.com/9193841/162201483-8afe157a-1ac0-4dc6-a865-37a8915e3c06.png)

If I wrote the query inverting the where region, I get the results in the reversed order:
![Screenshot 2022-04-07 at 14 47 23](https://user-images.githubusercontent.com/9193841/162202159-2f4f8044-5ab7-479b-aaa2-3fe431a50cd5.png)

Is a good idea depends on the where order to take the correct result or is an accidental behaviour that can change? Should I add an `order by` to ensure order?

**Comment by grisha-kotler** [comment, marked as answer]
You should use `order by` to ensure order

**Comment by ayende** [comment]
Just to note, you are writing this query:

```
from 'Core/Invoicing/TaxRule'
where Country == "ESP"
and Region == "Canarias"
or Region == null
limit 0, 1
```


But I don't think that this does what you want, you should use `()` to properly express this.

```
from 'Core/Invoicing/TaxRule'
where Country == "ESP"
and (Region == "Canarias" or Region == null)
limit 0, 1
```

**Reply by montyclt** [reply]
Thanks ayende, thats true, I wrote bad query.

How can I add the `()` on `DocumentQuery` API in C# client?

That is that I have now:
```cs
 query
    .WhereEquals("Region", region)
    .OrElse().WhereEquals("Region", null)
    .OrderByDescending("Region");
```

**Reply by krivulcik** [reply]
You can use `.OpenSubclause()` and `.CloseSubclause()`:

```c#
query
    .OpenSubclause()
    .WhereEquals("Region", region)
    .OrElse().WhereEquals("Region", null)
    .CloseSubclause()
    .OrderByDescending("Region");
```

**Reply by montyclt** [reply]
Thanks @krivulcik
